cadcdata/cadcdata/storageinv.py

```diff
@@ -0,0 +1,826 @@
+        '      cadcput -v -n cadc:TEST/ myfile1.fits.gz myfile2.fits.gz\n'
+        '- Use a different netrc file to put files from a directory to '
+        'directory dir:\n'
+        '      cadcput -d --netrc ~/mynetrc -s ivo:cadc.nrc.ca/cadc/minoc '
+        'cadc:TEST/ dir\n'
+        '- Connect as user to put files from multiple sources (prompt for\n'
+        '  password if user not in $HOME/.netrc):\n'
+        '      cadcput -v -u auser cadc:TEST/ myfile.fits.gz dir1 dir2')
+
+    args = parser.parse_args()
+    subject = net.Subject.from_cmd_line_args(args)
+
+    client = StorageInventoryClient(subject, args.service, host=args.host,
+                                    insecure=args.insecure)
+    files = []
+    for file in args.src:
+        if os.path.isdir(file):
+            # append all files in the directory
+            for i in os.listdir(file):
+                if not os.path.isdir(os.path.join(file, i)):
+                    files.append(os.path.join(file, i))
+        else:
+            files.append(file)
+    if (len(files) > 1) and not args.identifier.endswith('/'):
+        raise RuntimeError(
+            'A root identifier (ends in "/") is required to put multiple '
+            'files: {}'.format(args.identifier))
+
+    for file in files:
+        if len(files) > 1:
+            id = '{}/{}'.format(args.identifier.strip('/'),
+                                os.path.basename(file))
+        else:
+            id = args.identifier
+        execute_cmd(args, client.cadcput, {'id': id,
+                                           'src': file,
+                                           'file_type': args.type,
+                                           'file_encoding': args.encoding})
+
+
+def cadcget_cli():
+    parser = util.get_base_parser(subparsers=False,
+                                  version=version.version,
+                                  service=DEFAULT_RESOURCE_ID)
+
+    parser.description = (
+        'Download files from the CADC Storage Inventory into the current\n'
+        'directory unless overriden with the -o option.')
+
+    parser.add_argument(
+        '-o', '--output',
+        help='write to file or other directory instead of the current one.',
+        required=False)
+    parser.add_argument(
+        'identifier',
+        help='unique identifier (URI) given to the file in the CADC '
+             'Storage Inventory')
+    parser.epilog = (
+        'Examples:\n'
+        '- Anonymously download a file to current directory:\n'
+        '      cadcget gemini:GEMINI/00aug02_002.fits\n'
+        '- Use certificate to get a cutout and save it to a file:\n'
+        '      cadcget --cert ~/.ssl/cadcproxy.pem -o '
+        '/tmp/700000o-cutout.fits\n'
+        '        cadc:CFHT/700000o[1]\n')
+
+    args = parser.parse_args()
+    subject = net.Subject.from_cmd_line_args(args)
+    client = StorageInventoryClient(subject, args.service, host=args.host)
```

Comment:
Does this instantiation need to support `insecure` as well?

Reply:
Most definitely.